src/components/inputs/single/SingleJourney.tsx

```diff
@@ -1,44 +1,86 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Calender from '../../shared/Input/Calendar';
 import AutoComplete from '../../shared/Input/AutoComplete';
 import Timer from '../../shared/Input/Timer';
 import { Button as PRButton } from 'primereact/button';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { InputActionTypes } from '../../../redux/actions';
+import { IInputs } from '../../../lib/models/inputs';
+import { getInputs } from '../../../redux/selectors/inputs';
 
 function SingleJourney() {
     const dispatch = useDispatch();
+    const defaultInputs = useSelector(getInputs)
+    const [userSource, setUserSource] = useState<string>(defaultInputs.source.text);
+    const [userDestination, setUserDestination] = useState<string>(defaultInputs.destination.text);
+    const [userDate, setUserDate] = useState<Date | Date[]>(defaultInputs.departureDate);
+    const [userTime, setUserTime] = useState<Date | Date[]>(defaultInputs.departureTime);
+
+
+    const onSourceChange = (e: string) => {
```

Comment:
This looks good, but it has an issue. In functional components, creating functions like this, creates a new copy of this function with every re-render. So whenever your component re-renders, onSourceChange will get re-initialized, this becomes a performance issue if you have many such functions. Solution is to use useCallback or again since these are state change handlers, useReducer would suffice.